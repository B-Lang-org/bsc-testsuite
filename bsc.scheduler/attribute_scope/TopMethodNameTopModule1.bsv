interface Ifc;
   method Action mymethod();
endinterface

(* synthesize *)
module sysTopMethodNameTopModule1 (Ifc);
   Reg#(UInt#(16)) rg <- mkRegU;

   (* descending_urgency = "mymethod, r" *)
   rule r;
      rg <= rg + 1;
   endrule

   method Action mymethod();
      rg <= rg + 2;
   endmethod
endmodule
