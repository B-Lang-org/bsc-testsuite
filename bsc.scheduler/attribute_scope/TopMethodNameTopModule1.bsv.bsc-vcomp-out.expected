checking package dependencies
compiling TopMethodNameTopModule1.bsv
code generation for sysTopMethodNameTopModule1 starts
Verilog file created: sysTopMethodNameTopModule1.v
All packages are up to date.
